scheduler/src/cook/mesos.clj

```diff
@@ -319,12 +321,20 @@
                                  (or server-port server-https-port) \#
                                  (if server-port "http" "https") \#
                                  (java.util.UUID/randomUUID)))
-    (.autoRequeue leader-selector)
-    (.start leader-selector)
-    (log/info "Started the leader selector")
+    (if api-only?
+      ; If this is an api-only instance, we still want all the leader election configuration to be set,
+      ; without actually joing the election process.
+      ; This allows the instance to correctly forward client requests to the leader as necessary.
+      (log/info "Will not join leader election, running as api-only")
```

Comment:
Should we add the hostname here?

Reply:
It is already included in the logs we emit (all logs are tagged with a hostname, along with other common metadata).